libs/shared/ui/src/lib/components/button-primitive/button-primitive.tsx

```diff
@@ -0,0 +1,96 @@
+import { type VariantProps, cva } from 'class-variance-authority'
+import { type ComponentPropsWithoutRef, type ElementRef, forwardRef } from 'react'
+import { twMerge } from '@qovery/shared/util-js'
+
+const buttonVariants = cva(['inline-flex', 'items-center', 'shrink-0', 'font-medium', 'transition'], {
+  variants: {
+    variant: {
+      solid: [],
+      surface: [],
+      outline: [],
+    },
+    color: {
+      brand: [],
+      neutral: [],
+      green: [],
+    },
+    size: {
+      xs: ['text-xs', 'px-1', 'py-0.5'],
+      sm: ['text-xs', 'h-7', 'px-2'],
+      md: ['text-xs', 'h-8', 'px-3'],
+      lg: ['text-sm', 'h-10', 'px-4'],
+    },
+    radius: {
+      rounded: ['rounded'],
+      full: ['rounded-full'],
+    },
+    loading: {
+      true: ['relative', 'pointer-events-none'],
```

Comment:
This is not really button primitive responsibility.
As `ButtonPrimitive` takes `className`, it makes more sense to handle `loading` classes in `Button` like so:
```jsx
    <ButtonPrimitive {...props} ref={forwardedRef} className={twMerge(loading ? 'absolute pointer-events-none' : '', className )}>
      {loading && <LoaderSpinner className="absolute m-auto left-0 right-0" theme="dark" />}
      <span className={loading ? 'opacity-0' : undefined}>{props.children}</span>
    </ButtonPrimitive>
```

Reply:
https://github.com/Qovery/console/pull/995/commits/c89f10da11a622fb99100f65979c71dfb581d658